### ISSUE TITLE
Action does not work with `act`

### ISSUE THREAD

**Issue by mikenikles** [jetpack-io/devbox-install-action#10]
I'm working on a follow-up blog post to [Test GitHub Actions locally](https://www.mikenikles.com/blog/test-github-actions-locally), but with a focus on how to use `act` ([GitHub repo](https://github.com/nektos/act)) locally and Devbox on GitHub Actions.

The work-in-progress pull request is at https://github.com/mikenikles/devbox-github-actions/pull/1.

**Setup**
1. Clone the repo and change to the `configure-devbox-on-ci` branch
2. `devbox shell`
3. `devbox run act`

I run into two issues:
1) If I remove the `refresh-cli: true` input, `devbox run act` fails with `exec: "node": executable file not found in $PATH: unknown`
2) If I keep the `refresh-cli: true` input, the installation of the Devbox CLI is stuck with an interactive prompt "Nix is not installed. Devbox will attempt to install it. Press enter to continue or ctrl-c to exit." This is because of [this line](https://github.com/jetpack-io/devbox/blob/main/internal/nix/install.go#L127) in the installation script.

I'm opening this in case someone on the Devbox team has insights into how we can make this work, while in parallel I continue to investigate too.

**Comment by mikenikles** [comment]
I wonder if [this line](https://github.com/jetpack-io/devbox/blob/main/internal/nix/install.go#L127) should be as follows:

```diff
- if isatty.IsTerminal(os.Stdout.Fd()) {
+ if isatty.IsTerminal(os.Stdout.Fd()) || isatty.IsCygwinTerminal(os.Stdout.Fd()) {
```

This is a bit of a guess, based on the fact that `act` does that [here](https://github.com/nektos/act/blob/master/pkg/common/git/git.go#L235).

**Comment by LucilleH** [comment]
I haven't encountered/tried this combination of devbox > act > devbox scenario. What I've seen the most is testing locally with `devbox run` and CICD using the installer + running the same script, assuming the same script will produce the same result.

I'm not sure how act works, and whether it installs a fresh nix + devbox every time you run the CICD locally (I hope not, or else the experience would be very slow). Thoughts? @mikenikles

**Comment by mikenikles** [comment]
I'm on the same page in terms of `devbox run` locally and on CI / CD.

`act` emulates a GitHub actions environment locally within a Docker container. The benefit of that is you can iterate on your `.github/workflows/*.yml` files locally, without pushing changes to GitHub to kick off a new workflow run.

I was hoping to get this to work to showcase how someone can use a `devbox.json` for their dev environment, their CI environment, and also run the GitHub workflows locally with `act` to iterate quickly.

Caching the Devbox CLI works the same as it does in a GitHub workflow, so that helps with the experience locally.

This is a low priority issue from my point of view. I figured I'd open it to see if anyone on the team goes like "Ah... I think I have a fix for that" 😄. Happy to even close it and if I ever get back to using `act` in a project, I may re-test and re-open this issue if necessary.

**Comment by airtonix** [comment]
this is now a problem, because I cannot debug why a goreleaser run fails to run due to the devbox/installer inexplicably not making goreleaser available as a command.

having to do a release, wait for a build, discover that yet again the problem isn't fixed gets very very tiring.

**Comment by airtonix** [comment]
related: https://github.com/catthehacker/docker_images/issues/119

**Comment by LucilleH** [comment]
Hi @airtonix,

The error is caused by determinate systems nix installer trying to setup a systemd service for nix by default, but a docker image doesn't have systemd.

Can you try adding `skip-nix-installation: 'true'` to the devbox action like this:
```
   - name: setup jetpack devbox
      uses: jetpack-io/devbox-install-action@v0.7.0
      with:
          skip-nix-installation: 'true'
```

And add the following step before to the devbox step:
```
    - name: Install nix in container
      uses: DeterminateSystems/nix-installer-action@v4
      with:
        logger: pretty
        extra-conf: experimental-features = ca-derivations fetch-closure
        init: 'none'
```

**Comment by airtonix** [comment]
just checked that the image for act that I'm using has root access (some of them don't 🤷🏻 ).


![](https://assets-v2.lottiefiles.com/a/e15609a4-1187-11ee-a61e-43492082a309/PqK31QQUTR.gif)


it does.

**Comment by airtonix** [comment]
aha, i get further by running act like : 

```
  act push \
    --platform ubuntu-22.04=catthehacker/ubuntu:act-22.04 \
    --eventpath .actevent.json \
    --workflows .github/workflows/release.yml \
    --job Build
```

**Comment by airtonix** [comment]
and now another error... looks to be something something something github api no auth error

**Comment by airtonix** [comment]
ok updated command gets me past that: 

```
  act push \
    -s GITHUB_TOKEN="$(gh auth token)" \
    --platform ubuntu-22.04=catthehacker/ubuntu:act-22.04 \
    --eventpath .actevent.json \
    --workflows .github/workflows/release.yml \
    --job Build
```

and now i'm at the same error i get as when run in githubs infrastructure 👍🏻 

debugging time

**Comment by airtonix** [comment]
as usual... i'm the muppet.

original problem was that the command i'm supposed to run is `goreleaser`, not `gorelease`

**Comment by LucilleH** [comment]
@airtonix glad it all worked! 😃
